src/WebJobs.Script/Workers/Rpc/RpcWorkerChannel.cs

```diff
@@ -174,12 +174,19 @@ internal void WorkerInitResponse(RpcEvent initEvent)
 
         public void SetupFunctionInvocationBuffers(IEnumerable<FunctionMetadata> functions)
         {
-            _functions = functions;
+            var languageFunctions = new List<FunctionMetadata>();
+
             foreach (FunctionMetadata metadata in functions)
             {
-                _workerChannelLogger.LogDebug("Setting up FunctionInvocationBuffer for function:{functionName} with functionId:{id}", metadata.Name, metadata.FunctionId);
-                _functionInputBuffers[metadata.FunctionId] = new BufferBlock<ScriptInvocationContext>();
+                if (Utility.IsFunctionMetadataLanguageSupportedByWorkerRuntime(metadata, _workerConfig?.Description?.Language))
```

Comment:
@pragnagopa pointed out offline that this should be filtered upstream (the caller) and not in this channel here,. I will address that as well.